packages/app-store/zoomvideo/lib/VideoApiAdapter.ts

```diff
@@ -107,35 +108,27 @@ const zoomAuth = (credential: CredentialPayload) => {
       zoomRefreshedTokenSchema
     );
 
-    // TODO: If the new token is invalid, initiate the fallback sequence instead of throwing
-    // Expanding on this we can use server-to-server app and create meeting from admin calcom account
-
-    const oldCredential = await prisma.credential.findUniqueOrThrow({ where: { id: credential.id } });
-    const parsedKey = zoomTokenSchema.safeParse(oldCredential.key);
-    if (!parsedKey.success) {
-      return Promise.reject(new Error("Invalid credentials were saved in the DB"));
-    }
-
-    const key = parsedKey.data;
-    key.access_token = newTokens.access_token;
-    key.refresh_token = newTokens.refresh_token;
+    const key = credential.key as ZoomToken;
+    key.access_token = newTokens.access_token ?? key.access_token;
+    key.refresh_token = (newTokens.refresh_token as string) ?? key.refresh_token;
     // set expiry date as offset from current time.
-    key.expiry_date = Math.round(Date.now() + newTokens.expires_in * 1000);
+    key.expiry_date =
+      typeof newTokens.expires_in === "number"
+        ? Math.round(Date.now() + newTokens.expires_in * 1000)
+        : key.expiry_date;
     // Store new tokens in database.
-    await prisma.credential.update({ where: { id: credential.id }, data: { key } });
+    await prisma.credential.update({
+      where: { id: credential.id },
+      data: { key: { ...key, ...newTokens } },
+    });
     return newTokens.access_token;
   };
 
   return {
     getToken: async () => {
-      let credentialKey: ZoomToken | null = null;
-      try {
-        credentialKey = zoomTokenSchema.parse(credential.key);
-      } catch (error) {
-        return Promise.reject("Zoom credential keys parsing error");
-      }
```

Comment:
If the keys were malformed here it would also throw an error without the opportunity to fetch new tokens